swift/ql/src/queries/Security/CWE-135/StringLengthConflation.ql

```diff
@@ -11,5 +11,78 @@
  */
 
 import swift
+import codeql.swift.dataflow.DataFlow
+import DataFlow::PathGraph
 
-select "TODO"
+class StringLengthConflationConfiguration extends DataFlow::Configuration {
+  StringLengthConflationConfiguration() { this = "StringLengthConflationConfiguration" }
+
+  override predicate isSource(DataFlow::Node node, string flowstate) {
+    // result of a call to to `String.count`
+    exists(MemberRefExpr member |
+      member.getBaseExpr().getType().toString() = "String" and // TODO: use of toString
+      member.getMember().toString() = "count" and // TODO: use of toString
+      node.asExpr() = member and
+      flowstate = "String"
```

Comment:
It will be needed to track results in the `NSString` -> `Swift.String` direction separately from results in the `Swift.String` -> `NSString` direction.  The tests have plenty of examples in both directions, but as yet the query only supports the latter.